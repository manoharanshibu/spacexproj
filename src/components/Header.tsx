import React from 'react';
import { useHistory } from 'react-router-dom'

const Header = () => {
    const history = useHistory()
    return (
        <div className="header" onClick={() => history.push('/')}><div className="header-text">SPCACE-X</div></div>
    )
}

export default Header;