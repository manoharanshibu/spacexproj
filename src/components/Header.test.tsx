import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import Header from './Header';

describe(('Header Component'), () => {

    it('renders correctly', () => {
        const tree = renderer
            .create(<Header title='SPACE-X' />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render the title text', () => {
        render(<Header title='SPACE-X' />);
        const linkElement = screen.getByText(/SPACE-X/i);
        expect(linkElement).toBeInTheDocument();
    });

    it('should render heading', () => {
        render(<Header title='SPACE-X' />);
        const linkElement = screen.getByTestId('header');
        expect(linkElement).toBeInTheDocument();
    });

    it('title should be visible', () => {
        render(<Header title='SPACE-X' />);
        const linkElement = screen.getByTestId('header');
        expect(linkElement).toBeVisible();
    });
})
