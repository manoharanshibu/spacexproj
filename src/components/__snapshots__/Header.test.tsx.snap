// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component renders correctly 1`] = `
<div
  className="header"
  onClick={[Function]}
>
  <div
    className="header-text"
    data-testid="header"
  >
    SPACE-X
  </div>
</div>
`;
